packages/modeling/src/operations/extrusions/earcut/polygonTrees.js

```diff
@@ -0,0 +1,64 @@
+const geom2 = require('../../../geometries/geom2')
+const plane = require('../../../maths/plane')
+const vec2 = require('../../../maths/vec2')
+const vec3 = require('../../../maths/vec3')
+const calculatePlane = require('../slice/calculatePlane')
+const assignHoles = require('./assignHoles')
+
+/*
+ * This class maps a 3D polygon onto a 2D space using an orthonormal basis.
+ * It tracks the mapping so that points can be reversed back to 3D losslessly.
+ * Also constructs a polygon hierarchy which associates holes with their outer solids.
+ */
+class PolygonTrees {
```

Comment:
Still having trouble swallowing this class name… not sure there’s a tree here… more like what the comment says… hierarchy

Reply:
P.S. I hate name choosing

Reply:
Changed to PolygonHierarchy